### PR TITLE
[9] POST /todos

### PR DESCRIPTION
**Pull request by pdany1116**
#### Description

Create a todo via POST /todos.

## Steps to Test or Reproduce

1. Follow test steps in README.
2. All tests should pass.

### CODE REVIEW

File: app/controllers/todos_controller.rb [hunk 1]
```diff
@@ -14,4 +14,22 @@ def show
   rescue ActiveRecord::RecordNotFound
     render json: '', status: :not_found
   end
+
+  def create
+    todo = Todo.create!(create_todo_params)
+    todo.url = "/todos/#{todo.id}"
```
Comment (cazacutudor) [inline app/controllers/todos_controller.rb]: The URL should be computed in the serializer.

File: app/controllers/todos_controller.rb [hunk 2]
```diff
@@ -14,4 +14,22 @@ def show
...
+    todo.save!
+
+    render json: todo, status: :created
+  rescue ActionController::ParameterMissing, ActiveRecord::RecordInvalid
+    render json: '', status: :unprocessable_entity
```
Comment (cazacutudor) [inline app/controllers/todos_controller.rb:23]: We should respond with the validation errors

File: db/migrate/20220713104011_set_default_values.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class SetDefaultValues < ActiveRecord::Migration[7.0]
+  def change
+    change_table :todos, bulk: true do
+      change_column_default :todos, :completed, from: nil, to: false
```
Comment (cazacutudor) [inline db/migrate/20220713104011_set_default_values.rb]: I think change_column_default should work outside of change_table. What's the scope of these functions?

File: app/controllers/todos_controller.rb [hunk 4]
```diff
@@ -14,4 +14,22 @@ def show
   rescue ActiveRecord::RecordNotFound
     render json: '', status: :not_found
   end
+
+  def create
```
Comment (cazacutudor) [inline app/controllers/todos_controller.rb:18]: Integration specs missing for todo creation

File: features/support/todo_helpers.rb [hunk 5]
```diff
@@ -4,8 +4,8 @@ module TodoHelpers
   include FactoryBot::Syntax::Methods
 
   def create_todo
-    create(:todo)
-    @last_todo_id = Todo.last.id
+    post todos_path, attributes_for(:todo).slice(:title)
```
Comment (cazacutudor) [inline features/support/todo_helpers.rb]: Why we are not including the title?

**Review by cazacutudor — CHANGES_REQUESTED** [review]

File: db/migrate/20220713104011_set_default_values.rb (hunk 3 above)
Comment (pdany1116) [reply db/migrate/20220713104011_set_default_values.rb]: Rubocop was complaining about altering the table in multiple operations (here 2). By specifying `bulk: true` to `change_table`, these 2 operations will be transformed in a single alter operation.

File: app/controllers/todos_controller.rb (hunk 2 above)
Comment (pdany1116) [reply app/controllers/todos_controller.rb:23]: I thought about it. Should we do this in another ticket?

File: features/support/todo_helpers.rb (hunk 5 above)
Comment (pdany1116) [reply features/support/todo_helpers.rb]: We are including only the title. `slice` returns only specified keys from hash.

**Review by cazacutudor — APPROVED** [review]

**Pull request merged**